From db5c5f5347ee3a14a598a474a8d362ad569c2c77 Mon Sep 17 00:00:00 2001
From: Maher Khenissi <maherx.khenissi@intel.com>
Date: Thu, 3 Jan 2013 16:01:36 +0100
Subject: [PATCH 77/78] wlan: fix klocwork issues

BZ: 77624

Pointer 'ChaabiMacAddr' returned from call to function
'malloc' at line 260 may be NULL

Pointer 'result' returned from call to function
'malloc' at line 114 may be NULL

Change-Id: I5421526ae11ff68002d450b282a0cb923030d7a3
Signed-off-by: Maher Khenissi <maherx.khenissi@intel.com>
Reviewed-on: http://android.intel.com:8080/84747
Reviewed-by: Regairaz, Julie <julie.regairaz@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 wlan_prov/wlan_provisioning.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/wlan_prov/wlan_provisioning.c b/wlan_prov/wlan_provisioning.c
index bf069a4..244c795 100644
--- a/wlan_prov/wlan_provisioning.c
+++ b/wlan_prov/wlan_provisioning.c
@@ -114,7 +114,11 @@ static char *parse_uevent_file(char *path, char *pattern)
 			/* Pattern found */
 			result = (char *) malloc(strlen(line) -
 						strlen(pattern) + 1);
-			strlcpy(result, line + strlen(pattern), strlen(line) - strlen(pattern) + 1);
+			if (result) {
+				strlcpy(result, line + strlen(pattern), strlen(line) - strlen(pattern) + 1);
+			} else {
+				LOGE("%s: memory allocation failed\n", __func__);
+			}
 			goto close;
 		}
 	}
@@ -364,7 +368,8 @@ int main(int argc, char **argv)
 #endif
 		/* chaabi read error OR no chaabi support */
 		ChaabiMacAddr = (unsigned char *) malloc(MAC_ADDRESS_LEN);
-		memcpy(ChaabiMacAddr, NullMacAddr, MAC_ADDRESS_LEN);
+		if (ChaabiMacAddr)
+			memcpy(ChaabiMacAddr, NullMacAddr, MAC_ADDRESS_LEN);
 
 		LOGW("MAC not found");
 #ifdef BUILD_WITH_CHAABI_SUPPORT
-- 
1.7.9

